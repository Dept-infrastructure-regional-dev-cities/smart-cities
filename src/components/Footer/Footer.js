import React from 'react';

import style from './Footer.scss';

const Footer = () => (
  <footer className={style.wrapper}>
    <h1>Footer</h1>
  </footer>
);

export default Footer;
