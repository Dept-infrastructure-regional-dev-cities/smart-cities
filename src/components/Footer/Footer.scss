@import '~style/variables';

.wrapper {
  border-top: 6px solid $color-primary--600;
  padding: 25px 0;
}
