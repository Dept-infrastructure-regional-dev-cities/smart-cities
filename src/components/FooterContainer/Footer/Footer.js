import React from 'react';
import PropTypes from 'prop-types';
import style from './Footer.scss';
import getSubCategorySectionId from '../../../helpers/getSubCategorySectionId';
import {
  ELEMENT_IDS,
  LINKS,
  STRINGS,
} from '../../../constants';
import coaWhite from '../../../images/coa-white.svg';
import dtaLogo from './dtaLogo.svg';

const links = [
  {
    text: STRINGS.FEEDBACK_LINK,
    href: LINKS.FEEDBACK,
  },
  {
    text: STRINGS.PRIVACY_LINK,
    href: `${LINKS.ADDITIONAL_INFORMATION}#${ELEMENT_IDS.PRIVACY_POLICY}`,
  },
  {
    text: STRINGS.HELP_LINK,
    href: LINKS.CONTACT_US,
  },
  {
    text: STRINGS.ACCESSIBILITY_LINK,
    href: `${LINKS.ADDITIONAL_INFORMATION}#${ELEMENT_IDS.ACCESSIBILITY}`,
  },
  {
    text: STRINGS.SECURITY_LINK,
    href: `${LINKS.ADDITIONAL_INFORMATION}#${ELEMENT_IDS.SECURITY}`,
  },
  {
    text: STRINGS.CONTENT_CREDIT,
    href: `${LINKS.ADDITIONAL_INFORMATION}#${ELEMENT_IDS.CONTENT_CREDIT}`,
  },
];

const Footer = props => (
  <footer className={style.wrapper}>
    <div className={style.container}>
      <div className={style.logosWrapper}>
        <div className={style.coaWrapper}>
          <img
            className={style.coaLogo}
            src={coaWhite}
            alt="coat of arms"
          />
        </div>

        <div className={style.logoDivider} />

        <img
          className={style.dtaLogo}
          src={dtaLogo}
          alt="dta"
        />
      </div>

      <div className={style.wordsWrapper}>
        {!!props.category && (
          <div className={style.categoryDetailsWrapper}>
            <p className={style.categoryTitle}>
              {props.category.name}
            </p>

            <div className={style.subCategoryLinksWrapper}>
              {props.category.subCategories.map(subCategory => (
                <a
                  key={subCategory.name}
                  className={style.subCategoryLink}
                  href={`#${getSubCategorySectionId(subCategory.name)}`}
                >
                  {subCategory.name}
                </a>
              ))}
            </div>
          </div>
        )}

        <div className={style.staticLinksWrapper}>
          {links.map(link => (
            <a
              key={link.href}
              href={link.href}
              className={style.staticLink}
            >{link.text}</a>
          ))}
        </div>

        <div className={style.copyright}>
          {STRINGS.COPYRIGHT}
        </div>
      </div>
    </div>
  </footer>
);

Footer.propTypes = {
  category: PropTypes.shape({
    name: PropTypes.string.isRequired,
    subCategories: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
    })).isRequired,
  }),
};

export default Footer;
