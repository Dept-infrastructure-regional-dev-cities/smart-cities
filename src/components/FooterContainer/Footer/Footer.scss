@import '~style/variables';

.wrapper {
  padding: 50px 0 55px;
  background: $color-grey--800;
  color: $color-white;
}

.container {
  @include container;

  display: flex;
  flex-direction: column;
  align-items: center;

  @include for-desktop-up {
    flex-direction: row;
    align-items: flex-start;
  }
}

.logosWrapper {
  display: flex;
  align-items: center;
  flex: none;
  margin-bottom: 50px;

  @include for-desktop-up {
    margin-bottom: 0;
    margin-right: 80px;
  }
}

.coaWrapper {
  margin-right: 18px;
}

.coaLogo {
  width: 140px;
  height: auto;
}

.logoDivider {
  width: 2px;
  height: 46px;
  margin-right: 32px;
  background: $color-grey--700;
}

.dtaLogo {
  width: 70px;
  height: auto;
}

.wordsWrapper {
  flex: auto;
}

.categoryDetailsWrapper {
  border-bottom: 1px solid $color-grey--700;
  margin-bottom: 20px;
}

.categoryTitle {
  margin-bottom: 16px;
  font-size: 20px;
  font-weight: 700;
}

.subCategoryLinksWrapper {
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  margin-bottom: 23px;

  @include for-tablet-portrait-up {
    flex-direction: row;
  }
}

.subCategoryLink {
  margin-right: 32px;
  text-decoration: none;
  padding-bottom: 1px;
  border-bottom: 1px solid $color-highlight;
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    margin-bottom: 0;
  }
}

.staticLinksWrapper {
  padding-bottom: 24px;
  margin-bottom: 24px;
  border-bottom: 1px solid $color-grey--700;
}

.staticLink {
  margin-right: 16px;
  font-size: 14px;
  text-decoration: underline;

  @include for-desktop-up {
    &:nth-of-type(2) {
      margin-right: 46px;
    }
  }
}

.copyright {
  font-size: 14px;
  line-height: 1.2;
}
