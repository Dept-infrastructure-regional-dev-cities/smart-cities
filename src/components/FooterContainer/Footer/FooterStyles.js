import styled from 'styled-components';

export default {
  Wrapper: styled.footer`
    padding: 50px 0 55px;
    background: black;
    color: white;
  `,
  Container: styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
  `,
};
