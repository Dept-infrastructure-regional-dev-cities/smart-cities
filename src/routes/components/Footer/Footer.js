import React from 'react';

import style from './Footer.scss';

const Footer = () => {
  return (
    <footer className={style.Footer}>
      <h1>Footer</h1>
    </footer>
  )
};

export default Footer;
