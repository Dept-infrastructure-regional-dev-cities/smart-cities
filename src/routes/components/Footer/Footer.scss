@import '~style/helper';

.wrapper {
  border-top: 5.6px solid $color-primary--700;
  background: $color-grey--100;
  padding: 25px 0;
}
