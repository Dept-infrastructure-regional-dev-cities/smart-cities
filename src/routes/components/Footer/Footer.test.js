
/* global it, describe, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';


describe('(Components) Footer - footer ', () => {
  it('should match Snapshot', () => {
    const renderedValue =  renderer.create(<Footer />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
