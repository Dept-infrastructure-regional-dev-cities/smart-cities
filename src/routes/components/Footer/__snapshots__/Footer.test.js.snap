// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) Footer - footer  should match Snapshot 1`] = `
<footer
  className="Footer"
>
  <h1>
    Footer
  </h1>
</footer>
`;
